ui/pages/confirmations/components/confirm/row/dataTree.tsx

```diff
@@ -94,11 +94,14 @@ export const DataTree = ({
   >(undefined);
 
   useEffect(() => {
-    getTokenDecimalsOfDataTree(data).then((decimals) => {
+    const getDecimals = async () => {
+      const decimals = await getTokenDecimalsOfDataTree(data);
       if (typeof decimals === 'number') {
         setTokenContractDecimals(decimals);
```

Comment:
Minor, as maybe beyond the scope of the fix, but another plug for the `useAsyncResult` hook.

It embeds the state within it and includes a cleanup function.

So code could become:

```
...

const { value: decimalsResponse } = useAsyncResult(
    () => getTokenDecimalsOfDataTree(data),
    [data]
);

const tokenContractDecimals = typeof decimalsResponse === 'number' ?
    decimalsResponse : tokenDecimals;

...
```

Reply:
interesting. Looks like a handy, custom hook. However, when I replace the code with
```
  const { value: decimalsResponse } = useAsyncResult(
    () => getTokenDecimalsOfDataTree(data),
    [data],
  );

  const tokenContractDecimals =
    typeof decimalsResponse === 'number' ? decimalsResponse : 0;
```

the snapshot updates. Something in this implementation affects the result in the test. It doesn't appear to produce the intended decimals (2).
![CleanShot 2024-08-02 at 12 36 34@2x](https://github.com/user-attachments/assets/0ed2daaf-82b1-4640-a158-51b2f167abb2)

Reply:
cc: @jpuri

Reply:
ah. that was it! thanks @matthewwalsh0! 
updated the code